pennylane_lightning/core/_state_vector_base.py

```diff
@@ -136,10 +136,11 @@ def _apply_basis_state(self, state, wires, use_async: Optional[bool] = None):
             raise ValueError("BasisState parameter and wires must be of equal length.")
 
         # Return a computational basis state over all wires.
-        if use_async == None:
-            self._qubit_state.setBasisState(list(state), list(wires))
-        else:
-            self._qubit_state.setBasisState(list(state), list(wires), use_async)
+        self._select_setBasisState(state, wires, use_async)
+
+    @abstractmethod
+    def _select_setBasisState(state, wires, use_async):
+        """Select the proper setBasisState from the C++ backend"""
```

Comment:
To avoid confusion in the base class. It is better that each device define how to call `setBasisState`